lib/solidus_kustomer/client.rb

```diff
@@ -34,14 +34,70 @@ def create(kobject_klass, attributes:, customer:)
       response.success? || raise(SolidusKustomer::CreateError)
     end
 
-    # Finds the customer via the Solidus database id
+    # Finds a customer via their Spree::User record. If the user does not have a kustomer_id
+    # associated, it also updates the original record to cache the UUID.
     #
-    # @param user [Spree::User] the user whose data we're looking for
+    # @param user [Spree::User] the customer we're looking for
     #
-    # @return [Hash] the customer data
-    def find_customer_by_external_id(user)
+    # @return [Hash] the customer data or nil if not present
+    def find_customer(user)
+      return find_customer_by_uuid(user.kustomer_id) if user.kustomer_id
+
+      customer = find_customer_by_email(user.email) || find_customer_by_external_id(user.id)
+      user.update!(kustomer_id: customer['id']) if customer
+      customer
+    end
+
+    # Finds a customer UUID via their Spree::User record. If the user does not have a kustomer_id
+    # associated, it also updates the original record to cache the UUID.
+    #
+    # @param user [Spree::User] the customer's UUID we're looking for
+    #
+    # @return [String] the customer UUID or nil if not present
+    def find_customer_id(user)
+      return user.kustomer_id if user.kustomer_id
+
+      customer = find_customer(user)
+      user.update!(kustomer_id: customer['id']) if customer
```

Comment:
Is this really needed in the first place? You already do it when creating a customer.

Reply:
Same as one of my replies above, this is to cover also the case where a customer already exists in Solidus and in Kustomer, but Solidus does not have information about it. This could derive from the extension being installed on an already set up store or in case of a migration.